Handle extraction of field values for struct array attribute vectors.

toregge opened the pull request:
@geirst: please review

Commit:
Handle extraction of field values for struct array attribute vectors.

geirst commented on searchcore/src/tests/proton/attribute/document_field_extractor/document_field_extractor_test.cpp:
```diff
@@ -0,0 +1,380 @@
+    auto doc = f.makeDoc();
+    doc->setValue(f.tagField, IntFieldValue(200));
+    doc->setValue(f.nameField, StringFieldValue("name200b"));
+    DocumentFieldExtractor extractor(*doc);
```
Consider keeping extractor inside fixture (setup during makeDoc() and store doc instance as well) to simplify test setup. This will also simplify assertExtracted() signature.

geirst commented on searchcore/src/tests/proton/attribute/document_field_extractor/document_field_extractor_test.cpp:
```diff
@@ -0,0 +1,380 @@
+const ArrayDataType arrayTypeString(*DataType::STRING);
+const WeightedSetDataType weightedSetTypeInt(*DataType::INT, false, false);
+const WeightedSetDataType weightedSetTypeString(*DataType::STRING, false, false);
+const int32_t noInt(std::numeric_limits<int32_t>::min());
```
Consider using search::attribute::getUndefined() here.

geirst reviewed (commented):
Overall looks good, please have a look at my comments

Commit:
Rename tag field to weight field.
Keep document field extractor within fixture.
Use search::attriubute::getUndefined to specify undefined int32_t value.

geirst reviewed (approved):
👍
